client/settings/upe-toggle/provider.js

```diff
@@ -20,6 +20,18 @@ const UpeToggleContextProvider = ( { children, defaultIsUpeEnabled } ) => {
 	const [ status, setStatus ] = useState( 'resolved' );
 	const { invalidateResolutionForStoreSelector } = useDispatch( STORE_NAME );
 
+	// We're now allowing to update UPE enabled status through the settings REST
+	// API, for that reason we need to create a way to keep track of those changes
+	// in here.
+	const updateFlagLocally = useCallback(
```

Comment:
We should consider getting rid of this provider now it's possible to update the UPE enabled status through the settings endpoint. This way, we can have a single source of truth again. I will create an issue to address this in another PR.

Reply:
Created #2234.